internal/utils.go

```diff
@@ -0,0 +1,17 @@
+package internal
+
+const (
+	ArgoCDSecretTypeLabel   = "argocd.argoproj.io/secret-type"
+	ArgoCDSecretTypeCluster = "cluster"
+	AppSetKind              = "ApplicationSet"
+	AppSetAPIGroup          = "argoproj.io/v1alpha1"
+)
+
+func IsArgoCDCluster(labels map[string]string) bool {
```

Comment:
I'm not super clear on what this function achieves

Reply:
Looks at the Secret labels and returns true if it has the ArgoCD key and value. We don't care about all the secrets, we just care about the ones storing ArgoCD cluster information.

Reply:
ok, it makes sense. Worth renaming the function `IsArgoCDClusterSecret`?

Reply:
I'm not very confused by the name, so if you feel we should rename it I can do that.

Reply:
The function on its own did, but used in context it's actually fine.
```
	if !utils.IsArgoCDCluster(s.GetLabels()) {
		return nil
	}
```